### DISCUSSION TITLE
How to set up TLS properly?

### DISCUSSION THREAD

**Discussion by montoulieu** [vladmandic/sdnext#465] [Q&A]
Hey there, so I'm migrating my workflow over to this frontend and I'm having some issues getting TLS working like I had in Auto1111.

The auto TLS extension that I was using before appears to work at first and I can connect to the frontend through https, but when you try to edit any settings or generate images it gives an error in the top corner with the following:
`Something went wrong [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:997)`

I also attempted to create a self signed key/cert pair with openssl and add those using the command line args like so:
`--tls-keyfile "E:\AI\Stable Diffusion\automatic\webui.key" --tls-certfile "E:\AI\Stable Diffusion\automatic\webui.cert"`,
but that just causes a ton of errors in the cli on startup.

![image](https://user-images.githubusercontent.com/1077298/234106375-ae73e1f6-fd7f-4e94-ad98-82326c7acb3a.png)

Would love some help figuring out how to get his working as it's the only bottleneck preventing me from switching over.

Thanks!

**Comment by qpl23** [comment]
This is down to gradio, I think. See [this comment](https://github.com/gradio-app/gradio/issues/563#issuecomment-1428998192) and the rest of its thread.

Seems like a gradio fix is in the pipeline; being impatient and somewhat reckless, I just edited blocks.py manually along the lines suggested by the comment I linked.
